MultiParameterPlasticityStressUpdate corrected.  TensileStressUpdate added

WilkAndy opened the pull request:
Corrected smoothing in MultiParameterPlasticityStressUpdate.
    It is now produces a provably C2 and convex yield surface (previously i had taken a computational shortcut that was incorrect).  The flow directions also mean the return-map algorithm has a unique solution, under certain mild assumptions that must also be used in any non-smoothed formulations.
    Updated in-code doco to reflect the enhancements.
    
    Fixes #9477
    
    TensileStressUpdate added.  This implements Rankine plasticity in the StressUpdate formulation.
    
    Refs #9337

WilkAndy committed:
Corrected smoothing in MultiParameterPlasticityStressUpdate.
It is now produces a provably C2 and convex yield surface (previously i had taken a computational shortcut that was incorrect).  The flow directions also mean the return-map algorithm has a unique solution, under certain mild assumptions that must also be used in any non-smoothed formulations.
Updated in-code doco to reflect the enhancements.

Fixes #9477

TensileStressUpdate added.  This implements Rankine plasticity in the StressUpdate formulation.

Refs #9337

WilkAndy committed:
tests of TensileStressUpdate

Refs #9337

WilkAndy committed:
doco of TensileStressUpdate

Refs #9337

moosebuild commented:
Job [Documentation](https://www.moosebuild.org/job/102523/) on 66d01a0 wanted to post the following:

View the site [here](http://mooseframework.org/docs/PRs/9536/site)

WilkAndy commented:
@bwspenc , this commit contains TensileStressUpdate, which is a 3-parameter plasticity model.  Possibly Amit might be interested in it but i don't know his github handle.

bwspenc commented:
Pinging @jain651

WilkAndy commented:
Anyone like to review this?  @bwspenc , @sapitts ?

bwspenc commented:
@WilkAndy I'll look at it.

WilkAndy commented:
Thanks @bwspenc .  It's not really a huge commit.  Lots of tests and doco, and comparatively little code.

WilkAndy committed:
CappedMohrCoulombStressUpdate added

Fixes #9477

MultiParameterPlasticityStressUpdate: one-line documentation typo fixup
CappedDruckerPragerStressUpdate: one-line typo fixup relevant for models that harden dilation angle

WilkAndy committed:
tests of CappedMohrCoulomb

Fixes #9477

WilkAndy committed:
doco of CappedMohrCoulomb

Fixes #9477

WilkAndy committed:
Enhancements to CappedMohrCoulomb stress initialisation
More jacobian tests

WilkAndy commented:
@bwspenc - you'd better review before this commit tends to infinity!  There really isn't much code, but there are 75 more tests and quite thorough documentation, which makes the PR look rather huge.

moosebuild commented:
Job [Modules debug](https://www.moosebuild.org/job/104505/) on bb564f9 : invalidated by @WilkAndy

unrelated timeouts in Modules debug

bwspenc commented:
Sorry, @WilkAndy!  I have looked through it a little bit.  Before I go commenting on individual things, the first big comment that I have is that the documentation is not based on the Markdown system that we are now using for all documentation. Is there any reason it can't be based on that system?

WilkAndy commented:
@bwspenc - documentation - i'm not really ready for the new system yet.  i don't have time right now to learn how to do it.  I tried while at INL and the system wasn't quite ready for me, so i decided to wait for a few months before diving into it.  I plan to transfer all my doco (not just this Mohr-Coulomb stuff) into the new system next month.  It'll be a big job, but right now i've just got too many other things to do.

bwspenc commented on modules/tensor_mechanics/tests/capped_mohr_coulomb/random1.i:
```diff
@@ -0,0 +1,325 @@
+  [../]
+[]
+
+[Kernels]
```
We should switch to using the [Modules/TensorMechanics/Master] Action in all models.  It you haven't looked at it yet, it's really nice -- it can set up all of the standard stress output variables, and sets up the kernels, variables (optionally), and strain calculator.

bwspenc reviewed (approved):
Wow, the Mohr-Coulomb model is a beast!
I've made a few comments here about little things, but they are all things that can be addressed later.

WilkAndy commented:
Thanks for the review, comments and merge, @bwspenc 

andy
